### PR TITLE
Ddcnl 1559

### PR DESCRIPTION
**Pull request by Muhammad-Ali-Ahmad**

### CODE REVIEW

File: app/views/incomeTaxComparison/IncomeSummary.scala.html [hunk 1]
```diff
@@ -41,16 +41,16 @@ <h2 id="incomeSummaryHeading" class="heading-large">@{incomeSummaryHeading}</h2>
         @messages("tai.incomeTaxComparison.incomeTax.fromEmployer.description")
     </p>
 
-    <table class="table--font-reset flush--bottom responsive" id="incomeSummaryComparisonTable" role="grid">
+    <table class="table--font-reset flush--bottom responsive" id="incomeSummaryComparisonTable" role="table">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeSummary.scala.html:44]: Was a role of grid appropriate for this? Was the table interactive? If not do you need a role of table?

File: app/views/incomeTaxComparison/IncomeSummary.scala.html [hunk 2]
```diff
@@ -64,22 +64,22 @@ <h2 id="incomeSummaryHeading" class="heading-large">@{incomeSummaryHeading}</h2>
         @for((employmentIncomeSourceDetail, count) <- incomeSourceViewModel.employmentIncomeSourceDetail.zipWithIndex) {
 
             <tr>
-                <td scope="row" role="gridcell" class="table-title">
-                    <span class="table-heading" aria-hidden="true" id="employerName">
+                <td role="cell" class="table-title">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeSummary.scala.html:67]: Guessing this is related to the role=grid. You probably don't need to add role=cell

File: app/views/incomeTaxComparison/IncomeSummary.scala.html [hunk 3]
```diff
@@ -64,22 +64,22 @@ <h2 id="incomeSummaryHeading" class="heading-large">@{incomeSummaryHeading}</h2>
...
+                    <span class="table-heading employerName" aria-hidden="true">
                         @messages("tai.incomeTaxComparison.taxCodes.table.employmentsReaderText")
                     </span>
                     <span class="carry-over">@employmentIncomeSourceDetail.name</span>
                 </td>
 
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeSummary.scala.html:74]: Same as previous comment

File: app/views/incomeTaxComparison/IncomeSummary.scala.html [hunk 4]
```diff
@@ -64,22 +64,22 @@ <h2 id="incomeSummaryHeading" class="heading-large">@{incomeSummaryHeading}</h2>
...
                     <span class="table-heading" aria-hidden="true">
                         @Html(incomeSourceViewModel.htmlNonBroken(messages("tai.CurrentTaxYear")))
                         @Html(incomeSourceViewModel.htmlNonBroken(messages("tai.incomeTaxComparison.incomeTax.column1", Dates.formatDate(TaxYear().end))))
                     </span>
                     <span id="amount-cy-@{count}">@employmentIncomeSourceDetail.amountCY</span>
                 </td>
 
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeSummary.scala.html:82]: Same as previous comment

File: app/views/incomeTaxComparison/IncomeSummary.scala.html [hunk 5]
```diff
@@ -104,21 +104,21 @@ <h2 id="incomeSummaryHeading" class="heading-large">@{incomeSummaryHeading}</h2>
             @for((pensionIncomeSourceDetail, count) <- incomeSourceViewModel.pensionIncomeSourceDetail.zipWithIndex) {
 
                 <tr>
-                    <td scope="row" role="gridcell" class="table-title">
-                        <span class="table-heading" aria-hidden="true" id="employerName">
+                    <td role="cell" class="table-title">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeSummary.scala.html:107]: Same as previous comment

File: app/views/incomeTaxComparison/IncomeSummary.scala.html [hunk 6]
```diff
@@ -104,21 +104,21 @@ <h2 id="incomeSummaryHeading" class="heading-large">@{incomeSummaryHeading}</h2>
...
+                        <span class="table-heading employerName" aria-hidden="true">
                         @messages("tai.incomeTaxComparison.taxCodes.table.pensionsReaderText")
                         </span>
                         <span class="carry-over">@pensionIncomeSourceDetail.name</span>
                     </td>
-                    <td scope="row" role="gridcell" class="numeric">
+                    <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeSummary.scala.html:113]: Same as previous comment

File: app/views/incomeTaxComparison/IncomeSummary.scala.html [hunk 7]
```diff
@@ -104,21 +104,21 @@ <h2 id="incomeSummaryHeading" class="heading-large">@{incomeSummaryHeading}</h2>
...
                         <span class="table-heading" aria-hidden="true">
                             @Html(incomeSourceViewModel.htmlNonBroken(messages("tai.CurrentTaxYear")))
                             @Html(incomeSourceViewModel.htmlNonBroken(messages("tai.incomeTaxComparison.incomeTax.column1", Dates.formatDate(TaxYear().end))))
                         </span>
                         <span id="pension-amount-cy-@{count}">@pensionIncomeSourceDetail.amountCY</span>
                     </td>
 
-                    <td scope="row" role="gridcell" class="numeric">
+                    <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeSummary.scala.html:121]: Same as previous comment

File: app/views/incomeTaxComparison/IncomeTax.scala.html [hunk 8]
```diff
@@ -33,7 +33,7 @@ <h2 id="incomeTaxHeading" class="heading-large">
 
     <p>@messages("tai.incomeTaxComparison.incomeTax.description")</p>
 
-    <table class="table--font-reset flush--bottom responsive" role="grid">
+    <table class="table--font-reset flush--bottom responsive" role="table">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeTax.scala.html:36]: Same as other comments about table

File: app/views/incomeTaxComparison/IncomeTax.scala.html [hunk 9]
```diff
@@ -50,19 +50,19 @@ <h2 id="incomeTaxHeading" class="heading-large">
         </thead>
         <tbody>
             <tr role="row">
-                <td scope="row" role="gridcell" id="estimatedIncomeTaxComparisonTableRowDescription" class="table-title">
+                <td role="cell" id="estimatedIncomeTaxComparisonTableRowDescription" class="table-title">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeTax.scala.html:53]: Same as other comments

File: app/views/incomeTaxComparison/IncomeTax.scala.html [hunk 10]
```diff
@@ -50,19 +50,19 @@ <h2 id="incomeTaxHeading" class="heading-large">
...
                     @messages("tai.incomeTaxComparison.incomeTax.estimate")
                 </td>
 
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeTax.scala.html:57]: Same as other comments

File: app/views/incomeTaxComparison/IncomeTax.scala.html [hunk 11]
```diff
@@ -50,19 +50,19 @@ <h2 id="incomeTaxHeading" class="heading-large">
...
                     <span class="table-heading" aria-hidden="true">
                         @Html(viewmodel.htmlNonBroken(messages("tai.CurrentTaxYear")))
                         @Html(viewmodel.htmlNonBroken(messages("tai.incomeTaxComparison.incomeTax.column1", viewmodel.currentTaxYearHeader)))
                     </span>
                     @pounds(viewmodel.comparisonItemsByYear.head.estimatedIncomeTax)
                 </td>
 
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/IncomeTax.scala.html:65]: Same as other comments

File: app/views/incomeTaxComparison/TaxCodes.scala.html [hunk 12]
```diff
@@ -33,27 +33,27 @@ <h2 class="heading-large" id="taxCodesHeading">@messages("tai.incomeTaxCompariso
     </p>
 }
 
-<table class="table--font-reset flush--bottom responsive" id="taxCodeComparisonTable" role="grid">
+<table class="table--font-reset flush--bottom responsive" id="taxCodeComparisonTable" role="table">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxCodes.scala.html:36]: Same as other table comments

File: app/views/incomeTaxComparison/TaxCodes.scala.html [hunk 13]
```diff
@@ -33,27 +33,27 @@ <h2 class="heading-large" id="taxCodesHeading">@messages("tai.incomeTaxCompariso
...
         </th>
     </tr>
     </thead>
-    <tbody id="taxCodeComparisonTableBody">
+    <tbody class="taxCodeComparisonTableBody">
         @for(taxCodeDetail <- model.employmentTaxCodes) {
             <tr role="row">
-                <td scope="row" role="gridcell" class="table-title">
-                    <span class="table-heading" aria-hidden="true" id="employerName">
+                <td role="cell" class="table-title">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxCodes.scala.html:55]: Same as other comments

File: app/views/incomeTaxComparison/TaxCodes.scala.html [hunk 14]
```diff
@@ -63,8 +63,8 @@ <h2 class="heading-large" id="taxCodesHeading">@messages("tai.incomeTaxCompariso
         }
         @for(taxCodeDetail <- model.pensionTaxCodes) {
             <tr role="row">
-                <td scope="row" role="gridcell" class="table-title">
-                    <span class="table-heading" aria-hidden="true" id="employerName">
+                <td role="cell" class="table-title">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxCodes.scala.html:66]: Same as other comments

File: app/views/incomeTaxComparison/TaxCodes.scala.html [hunk 15]
```diff
@@ -77,7 +77,7 @@ <h2 class="heading-large" id="taxCodesHeading">@messages("tai.incomeTaxCompariso
 </table>
 
 @tableRow(taxCodeDetail: TaxCodeDetail) = {
-    <td scope="row" role="gridcell" class="numeric">
+    <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxCodes.scala.html:80]: Same as other comments

File: app/views/incomeTaxComparison/TaxCodes.scala.html [hunk 16]
```diff
@@ -86,7 +86,7 @@ <h2 class="heading-large" id="taxCodesHeading">@messages("tai.incomeTaxCompariso
         @taxCodeDetail.taxCodes.head
     </td>
 
-    <td scope="row" role="gridcell" class="numeric">
+    <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxCodes.scala.html:89]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 17]
```diff
@@ -35,28 +35,28 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
 <p>@messages("tai.incomeTaxComparison.taxFreeAmount.PA.information2")</p>
 
 
-<table class="table--font-reset flush--bottom responsive" role="grid">
+<table class="table--font-reset flush--bottom responsive" role="table">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:38]: Same as other table comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 18]
```diff
@@ -35,28 +35,28 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
     </thead>
     <tbody>
         <tr>
-            <td scope="row" class="tax-free-amount-table-heading">
+            <td  class="tax-free-amount-table-heading">
                 @messages("tai.income.personalAllowance")
             </td>
             @for((personalAllowance, itemIndex) <- model.personalAllowance.values zip (Stream from 1)) {
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:59]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 19]
```diff
@@ -76,17 +76,17 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
 </table>
 
 
-<table class="table--font-reset flush--bottom my-new-class responsive" role="grid">
+<table class="table--font-reset flush--bottom my-new-class responsive" role="table">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:79]: Same as other table comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 20]
```diff
@@ -96,11 +96,11 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
     <tbody>
         @for(row <- model.additions.additions) {
             <tr>
-                <td scope="row" role="gridcell" class="table-title">@messages("tai.taxFreeAmount.table.taxComponent." + row.label)</td>
+                <td role="cell" class="table-title">@messages("tai.taxFreeAmount.table.taxComponent." + row.label)</td>
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:99]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 21]
```diff
@@ -96,11 +96,11 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
                 @for((addition, itemIndex) <- row.values zip (Stream from 1)) {
                     @{itemIndex match {
                         case 1 =>
-                            <td scope="row" role="gridcell" class="numeric">
+                            <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:103]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 22]
```diff
@@ -109,7 +109,7 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
                             </td>
 
                         case 2 =>
-                            <td scope="row" role="gridcell" class="numeric">
+                            <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:112]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 23]
```diff
@@ -124,19 +124,19 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
 
         @if(!model.hasAdditions) {
             <tr>
-                <td scope="row" role="gridcell" class="table-title">
+                <td role="cell" class="table-title">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:127]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 24]
```diff
@@ -124,19 +124,19 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
                     <span aria-hidden="true">
                     @messages("tai.incomeTaxComparison.taxFreeAmount.noAdditions")
                     </span>
                 </td>
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:132]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 25]
```diff
@@ -124,19 +124,19 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
                     <span class="table-heading" aria-hidden="true">
                         @Html(model.htmlNonBroken(messages("tai.CurrentTaxYear")))
                         @Html(model.htmlNonBroken(messages("tai.incomeTaxComparison.incomeTax.column1", Dates.formatDate(TaxYear().end))))
                     </span>
                     @pounds(0)
                 </td>
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:139]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 26]
```diff
@@ -151,16 +151,16 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
 </table>
 
 
-<table class="table--font-reset flush--bottom responsive" role="grid">
+<table class="table--font-reset flush--bottom responsive" role="table">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:154]: Same as other table comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 27]
```diff
@@ -169,11 +169,11 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
     <tbody>
         @for(row <- model.deductions.deductions) {
             <tr>
-                <td scope="row" role="gridcell" class="table-title">@messages("tai.taxFreeAmount.table.taxComponent." + row.label)</td>
+                <td role="cell" class="table-title">@messages("tai.taxFreeAmount.table.taxComponent." + row.label)</td>
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:172]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 28]
```diff
@@ -169,11 +169,11 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
                 @for((deduction, itemIndex) <- row.values zip (Stream from 1)) {
                     @{itemIndex match {
                         case 1 =>
-                            <td scope="row" role="gridcell" class="numeric">
+                            <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:176]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 29]
```diff
@@ -182,7 +182,7 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
                             </td>
 
                         case 2 =>
-                            <td scope="row" role="gridcell" class="numeric">
+                            <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:185]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 30]
```diff
@@ -198,19 +198,19 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
 
         @if(!model.hasDeductions) {
             <tr>
-                <td scope="row" role="gridcell" class="table-title">
+                <td role="cell" class="table-title">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:201]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 31]
```diff
@@ -198,19 +198,19 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
                     <span aria-hidden="true">
                     @messages("tai.incomeTaxComparison.taxFreeAmount.noDeductions")
                     </span>
                 </td>
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:206]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 32]
```diff
@@ -198,19 +198,19 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
                     <span class="table-heading" aria-hidden="true">
                         @Html(model.htmlNonBroken(messages("tai.CurrentTaxYear")))
                         @Html(model.htmlNonBroken(messages("tai.incomeTaxComparison.incomeTax.column1", Dates.formatDate(TaxYear().end))))
                     </span>
                     @pounds(0)
                 </td>
-                <td scope="row" role="gridcell" class="numeric">
+                <td role="cell" class="numeric">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:213]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 33]
```diff
@@ -224,7 +224,7 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
 
 </table>
 
-<table class="table--font-reset table--spaced-top responsive">
+<table class="table--font-reset table--spaced-top responsive" role="table">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:227]: Same as other table comments

File: app/views/incomes/yourIncomeCalculation.scala.html [hunk 34]
```diff
@@ -124,6 +116,14 @@ <h2 class="heading-medium">@Messages("tai.income.calculation.heading",s"${Dates.
                                     </tr>
                                 }
                             </tbody>
+                            <tfoot>
```
Comment (timsb) [inline app/views/incomes/yourIncomeCalculation.scala.html:119]: This change isn't needed. <tfoot> before <tbody> is valid and can sometimes be an improvement

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 35]
```diff
@@ -244,14 +244,14 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
             <td>@messages("tai.incomeTaxComparison.taxFreeAmount.totalTFA")</td>
             @for((total, itemIndex) <- model.footer.values zip (Stream from 1)) {
                 @if(itemIndex ==1){
-                    <td scope="row" role="gridcell" class="numeric font-small">
+                    <td role="cell" class="numeric font-small">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:247]: Same as other comments

File: app/views/incomeTaxComparison/TaxFreeAmount.scala.html [hunk 36]
```diff
@@ -244,14 +244,14 @@ <h2 class="heading-large" id="taxFreeAmountHeading">@messages("tai.incomeTaxComp
...
                         <span class="table-heading" aria-hidden="true">
                             @Html(model.htmlNonBroken(messages("tai.CurrentTaxYear")))
                         </span>
                         @pounds(total)
                     </td>
                 }else {
-                    <td scope="row" role="gridcell" class="numeric font-small">
+                    <td role="cell" class="numeric font-small">
```
Comment (timsb) [inline app/views/incomeTaxComparison/TaxFreeAmount.scala.html:254]: Same as other comments

File: app/uk/gov/hmrc/tai/viewModels/estimatedIncomeTax/DetailedIncomeTaxEstimateViewModel.scala [hunk 37]
```diff
@@ -279,18 +279,22 @@ object DetailedIncomeTaxEstimateViewModel extends BandTypesConstants with Income
...
       .map(_.amount)
       .getOrElse(BigDecimal(0))
+
+    val tabIndexLink = {
+      val link: Link = Link
+        .toInternalPage(
+          url = routes.YourTaxCodeController.taxCodes.url,
+          value = Some(Messages("tai.taxCollected.atSource.marriageAllowance.description.linkText"))
+        )
+      val tabLink = link.copy(dataAttributes = Some(Map("taxindex" -> "-1")))
```
[the remark this answers is not in the document]
Comment (Muhammad-Ali-Ahmad) [reply app/uk/gov/hmrc/tai/viewModels/estimatedIncomeTax/DetailedIncomeTaxEstimateViewModel.scala:289]: good spot

**Pull request merged**